File: 프로그래머스연습문제/정수 내림차순으로 배치하기.cpp
#include <string>
#include <vector>
#include <algorithm>
#include <cmath>
using namespace std;

bool compare(int a, int b){
    
    return b < a;
}

long long solution(long long n) {
    long long answer = 0;
    vector<long long int> number;
    
    while (n>=10){
        number.push_back(n%10);
        n = n/10;
    }
    number.push_back(n);
    
    sort(number.begin(), number.end(),compare);
    
    for(long long int i = 0; i < number.size(); i++){
        answer += number[i] * pow(10,number.size()-i-1);
    }
    
    return answer;
}

File: 프로그래머스_python/연습문제/Level1/정수 내림차순으로 배치하기.py
def solution(n):
    answer = 0
    
    array = []
    
    while (n > 0):
        
        array.append(n%10)
        n //= 10
    
    array = sorted(array,reverse = True)
    
    for i in range(len(array)):
        answer = answer * 10 + array[i]
        
    
    return answer
